### PR TITLE
Added support for noisy dense layer

### PR DESCRIPTION
**Pull request by LeonShams**
Added support for noisy dense layers which are just dense layers with some random noise (decayed through gradient descent) added to the weights matrix to help with exploration in reinforcement learning environments.

Paper on noisy nets: https://arxiv.org/pdf/1706.10295.pdf

### CODE REVIEW

**Comment by LeonShams** [comment]
@googlebot I signed it!

**Comment by LeonShams** [comment]
@pavithrasv Please let me know if there are any problems or if you have any questions. Thanks!

**Comment by pavithrasv** [comment]
Should we add this to TF add ons? https://github.com/tensorflow/addons/tree/master/tensorflow_addons/layers

**Comment by LeonShams** [comment]
I just assumed that because this was so similar to the traditional dense layer and because it is very commonly used in RL that it should be added to TensorFlow, but if you think it would fit TF add ons better then I can submit a PR there.

**Comment by pavithrasv** [comment]
Yes, I think this will be a better fit for the addons repo. This layer is similar to having Dense + Dropout. What we recommend for new layers is that, they be added to the addons repo and if we see a lot of requests for the layer to be in the core APIs, we can move them to core.

**Comment by LeonShams** [comment]
Ok then. I'll submit a PR to TF add ons.

**Comment by pavithrasv** [comment]
Thank you!
